src/black/files.py

```diff
@@ -339,7 +348,8 @@ def gen_python_files(
 
     assert root.is_absolute(), f"INTERNAL ERROR: `root` must be absolute but is {root}"
     for child in paths:
-        root_relative_path = child.absolute().relative_to(root).as_posix()
+        assert child.is_absolute()
```

Comment:
Not new, this should be guaranteed by `path = root / (path.resolve().relative_to(root))` in `get_sources`